src/app/util/af-event.cpp

```diff
@@ -124,7 +170,8 @@ EmberStatus emberAfEventControlSetDelayMS(EmberEventControl * control, uint32_t
 {
     if (delayMs <= EMBER_MAX_EVENT_CONTROL_DELAY_MS)
     {
-        // TODO: set a CHIP timer here emberEventControlSetDelayMS(*control, delayMs); #658
+        control->status = EMBER_EVENT_MS_TIME;
+        chip::DeviceLayer::SystemLayer.StartTimer(delayMs, EventControlHandler, control);
```

Comment:
So `StartTimer` with a 0 delay automatically does the same thing as `ScheduleWork`?

Reply:
I can not really confirm that. I believe this is platform dependent.

Reply:
OK.  The original impl of `emberAfEventControlSetDelayMS` looks like this:
```
EmberStatus emberAfEventControlSetDelayMS(EmberEventControl *control,
                                          uint32_t delayMs)
{
  if (delayMs == 0) {
    emberEventControlSetActive(*control);
  } else if (delayMs <= EMBER_MAX_EVENT_CONTROL_DELAY_MS) {
    emberEventControlSetDelayMS(*control, delayMs);
```
so if we want to preserve those semantics and don't know whether `StartTimer` guarantees them, I think we should introduce that check-for-0 thing.

Reply:
I can confirm that starttimer() of zero runs the timer immediately.  Also: SetActive() calling the handler immediately can cause a stack overflow (i.e. where running the handler merely reschedules the handler with 0).

Reply:
schedulework() looks exactly like a zero-based timer without the reverse sort.  it's not really platform-dependent, as all the lists are in the CHIP system layer.